tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Parsing/CodeownersEntryPathComparer.cs

```diff
@@ -0,0 +1,254 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Sdk.Tools.CodeownersUtils.Parsing
+{
+    /// <summary>
+    /// Custom comparer for CodeownersEntry that sorts by path alphabetically,
+    /// with wildcard paths appearing before non-wildcard paths when they share the same base.
+    /// </summary>
+    public class CodeownersEntryPathComparer : IComparer<CodeownersEntry>
+    {
+        public int Compare(CodeownersEntry x, CodeownersEntry y)
+        {
+            if (x == null && y == null) return 0;
+            if (x == null) return -1;
+            if (y == null) return 1;
```

Comment:
```suggestion 
            if (x == null && y == null) {
                return 0;
            }
            if (x == null) {
                return -1;
            }
            if (y == null) {
                return 1;
            }
```

Avoid braceless statements as they can be a source of nonintuitive runtime errors in the future https://www.imperialviolet.org/2014/02/22/applebug.html